src/coreclr/jit/fgbasic.cpp

```diff
@@ -955,35 +1000,126 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr, IL_OFFSET codeSize, Fixed
 
                 CORINFO_METHOD_HANDLE methodHnd      = nullptr;
                 bool                  isJitIntrinsic = false;
-                bool                  mustExpand     = false;
                 NamedIntrinsic        ni             = NI_Illegal;
 
                 if (resolveTokens)
                 {
+                    assert(makeInlineObservations);
                     impResolveToken(codeAddr, &resolvedToken, CORINFO_TOKENKIND_Method);
                     methodHnd      = resolvedToken.hMethod;
                     isJitIntrinsic = eeIsJitIntrinsic(methodHnd);
                 }
 
                 if (isJitIntrinsic)
                 {
-                    intrinsicCalls++;
                     ni = lookupNamedIntrinsic(methodHnd);
 
+                    bool foldableIntrinsc = false;
+
+                    if (IsMathIntrinsic(ni) && FgStack::IsConstant(pushedStack.Pop()))
```

Comment:
Notably certain intrinsics are only foldable when we aren't in prejit scenarios due to xplat differences.

Others, like `sqrt`, are always emitted as an instruction rather than a call (we have some existing helper methods for differentiating some of these).